control/setup_control.go

```diff
@@ -0,0 +1,408 @@
+package control
+
+import (
+	"context"
+	"sync"
+
+	"go.uber.org/multierr"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/logging"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+// rPiGain is 1/255 because the PWM signal on a pi (and most other boards)
+// is limited to 8 bits, or the range 0-255.
+const rPiGain = 0.00392157
+
+var (
+	// default derivative type is "backward1st1".
+	derivativeType   = "backward1st1"
+	loopFrequency    = 50.0
+	sumIndex         = 1
+	linearPIDIndex   = 2
+	angularPIDIndex  = -1
+	controllableType = "motor_name"
+)
+
+// PIDLoop is used for setting up a PID control loop.
+type PIDLoop struct {
+	BlockNames              map[string][]string
+	ControlConf             Config
+	ControlLoop             *Loop
```

Comment:
We should keep an eye on whether passing this through a pointer makes sense. 
https://medium.com/@meeusdylan/when-to-use-pointers-in-go-44c15fe04eac